src/libaudqt/fileopener.cc

```diff
@@ -116,7 +172,11 @@ EXPORT void fileopener_show(FileMode mode)
                     break;
                 case FileMode::ExportPlaylist:
                     if (files.len() == 1)
-                        export_playlist(playlist, files[0].filename);
+                    {
+                        QString filter = dialog->selectedNameFilter();
+                        String extension = get_extension_from_filter(filter);
+                        export_playlist(playlist, files[0].filename, extension);
```

Comment:
With the changes suggested above, this would become:
```
                        QString extension = get_extension_from_filter(filter);
                        export_playlist(playlist, files[0].filename,
                                        extension.toUtf8());
```